File: Numerical/Gauss - Jordan.cpp
#include<stdio.h>
#include<iostream>
#include<ctype.h>
#include<string.h>
#include<stdlib.h>
#include<math.h>
#include<string>
#include<algorithm>
#include<vector>
#include<sstream>
#include<stack>
#include<queue>
#include<map>
#include<set>
using namespace std;

//const double PI=3.14159265358979323846264338327950288419716939937511;
double arr[10][10],pivot;
int N,i,j,k,row,col;
int main()
{
    freopen("input.txt","r",stdin);
    //freopen("output.txt","w",stdout);
    while(scanf("%d",&N)==1){
        row=N;
        col=N+1;
        for(i=1;i<=row;i++){
            for(j=1;j<=col;j++){
                scanf("%lf",&arr[i][j]);
            }
        }
        for(k=1;k<=row;k++){
            pivot=arr[k][k];
            for(i=1;i<=row;i++){
                if(i==k) continue;
                double pivotdown=arr[i][k];
                for(j=1;j<=col;j++){
                    //printf("(%d,%d) = %.2lf(%.2lf) - %.2lf(%.2lf)",i,j,pivot,arr[i][j],pivotdown,arr[k][j]);
                    arr[i][j]=pivot*arr[i][j]-pivotdown*arr[k][j]; // r2/r3.. = pivot r2/r3/.. - pivotdown*pivotrow
                    //printf(" = %.2lf\n",arr[i][j]);
                }
                //puts("");
            }
        }/*
        for(i=1;i<=row;i++){
            for(j=1;j<=col;j++){
                printf("%lf ",arr[i][j]);
            }
            puts("");
        }*/
        for(i=1,j=1;i<=row;i++,j++){
            pivot=arr[i][j];
            arr[i][j]=arr[i][j]/pivot;
            arr[i][col]=arr[i][col]/pivot;
        }
        for(i=1;i<=row;i++){
            printf("%.2lf\n",arr[i][col]);
        }
        puts("");
    }
    return 0;
}

File: Numerical/gauss jordan.m
clear
row=input('input dimension');
for i=1:row
    for j=1:row
        arr(i,j)=input('input value ');
    end
    arrb(i)=input('input value ');
end
for i=1:row
    c=abs(arr(i,i));
    d=i;
    for j=i:row
        if abs(arr(j,i))>c
            c=abs(arr(j,i));
            d=j;
        end
    end
    if(c==0)
        break
    end
    for j=1:row
        e=arr(i,j);
        arr(i,j)=arr(d,j);
        arr(d,j)=e;
    end
    e=arrb(i);
    arrb(i)=arrb(d);
    arrb(d)=e;
    c=arr(i,i);
    e=(c-1)/c;
    for j=1:row
        arr(i,j)=arr(i,j)-e*arr(i,j);
    end
    arrb(i)=arrb(i)-e*arrb(i);
    c=arr(i,i);
    for j=1:row
        if(j==i)
            continue
        end
        d=arr(j,i)/c;
        for k=1:row
            arr(j,k)=arr(j,k)-d*arr(i,k);
        end
        arrb(j)=arrb(j)-d*arrb(i);
    end
    arr
end
if (c==0)
    disp('Not Possible')
else    
    b
end